src/elektra/redshift.ini

```diff
@@ -0,0 +1,148 @@
+; Based on: https://github.com/ElektraInitiative/libelektra/blob/master/examples/spec/lcdproc/LCDd.ini
+
+; TODO:
+; 6. add command line arguments to specification and split up colon-separated options for location-providers and adjustment-methods into separate CLI arguments.
+
+[]
+mountpoint = redshift.ecf
+infos/plugins = ni validation
+
+; Configuration values - section redshift
+;=======================================================
+[temp/day]
+type = unsigned_short
+description = The color temperature the screen should have during daytime.
+default = 6500
+
+[temp/night]
+type = unsigned_short
+description = The color temperature the screen should have during nighttime.
+default = 4500
+
+[fade]
+type = boolean
+description = Whether transitions between color temperatures (e.g. from daytime to nighttime) should happen fast or slow.
+default = 1
+
+[brightness]
+type = float
+description = The screen brightness.
+default = 1.0
+check/range = 0-1
+
+[brightness/day]
+type = float
+description = The screen brightness during daytime.
+default = 1.0
+check/range = 0-1
+
+[brightness/night]
+type = float
+description = The screen brightness during nighttime.
+default = 1.0
+check/range = 0-1
+
+[elevation/high]
+type = float
+description = By default, Redshift will use the current elevation of the sun to determine whether it is daytime, night or in transition (dawn/dusk). When the sun is above the degrees specified with the elevation-high key it is considered daytime and below the elevation-low key it is considered night (source: https://github.com/jonls/redshift/wiki/Configuration-file#solar-elevation-thresholds).
+default = 3.0
+
+[elevation/low]
+type = float
+description = By default, Redshift will use the current elevation of the sun to determine whether it is daytime, night or in transition (dawn/dusk). When the sun is above the degrees specified with the elevation-high key it is considered daytime and below the elevation-low key it is considered night (source: https://github.com/jonls/redshift/wiki/Configuration-file#solar-elevation-thresholds).
+default = -6.0
+
+[gamma]
+type = float
+description = The screen gamma.
+default = 1.0
+
+[gamma/day]
+type = float
+description = The screen gamma during daytime.
+default = 1.0
+
+[gamma/night]
+type = float
+description = The screen gamma during nighttime.
+default = 1.0
+
+[adjustment-method]
+type = string
+check/enum = #5
+check/enum/#0 = drm
+check/enum/#1 = dummy
+check/enum/#2 = quartz
+check/enum/#3 = randr
+check/enum/#4 = vidmode
+check/enum/#5 = w32gdi
+description = The method used to adjust screen color temperature.
+require =
+
+[location-provider]
+type = string
+check/enum = #2
+check/enum/#0 = corelocation
+check/enum/#1 = geoclue2
+check/enum/#2 = manual
+description = The location provider. Used to establish whether it is currently daytime or nighttime.
+require =
+
+[dawn-time]
+type = string
+check/date = ISO8601
+check/date/format = timeofday
```

Comment:
Please give example and default.

Reply:
This is an optional value. If it set, solar elevation is ignored and the time is used instead. Therefore, setting it, has side-effects.

To make that more clearer, I'll add example and default and also add another value `[time/enabled]` that defaults to false.

Reply:
Please also think about if there is a way to completely avoid such effects with a more elegant design of the config.

Reply:
I've restructured the hierarchy according to your suggestion!

I've added a key [provider] where one can pick location vs. time and moved the time keys into [provider/time].